File: code/generate_strategies/IR(CoT)/java/java/function_gensim_models_Word2Vec.java
package com.example;

import java.util.*;
import java.util.stream.Collectors;

import org.deeplearning4j.models.word2vec.Word2Vec;
import org.deeplearning4j.text.sentenceiterator.CollectionSentenceIterator;
import org.deeplearning4j.text.tokenization.tokenizerfactory.DefaultTokenizerFactory;

public class Word2VecModelTrainer {

    public String train_word2vec_model(String sentences_str) {
        try {
            
            List<String> sentenceList = Arrays.stream(sentences_str.split(";"))
                    .map(String::trim)
                    .filter(s -> !s.isEmpty())
                    .collect(Collectors.toList());

            
            if(sentenceList.isEmpty()){
                return "Training failed";
            }

            
            CollectionSentenceIterator sentenceIterator = new CollectionSentenceIterator(sentenceList);
            
            DefaultTokenizerFactory tokenizerFactory = new DefaultTokenizerFactory();

            
            Word2Vec vec = new Word2Vec.Builder()
                    .minWordFrequency(1)
                    .iterations(1)
                    .layerSize(50)
                    .windowSize(3)
                    .seed(42)
                    .iterate(sentenceIterator)
                    .tokenizerFactory(tokenizerFactory)
                    .workers(4)
                    .build();
            vec.fit();

            
            List<String> vocab = new ArrayList<>(vec.getVocab().words());
            
            return String.join("|", vocab);
        } catch(Exception e) {
            return "Training failed";
        }
    }
}

File: manual/code/generate_strategies/IR(pseudocode)/python/python/function_gensim_models_Word2Vec.py
from gensim.models import Word2Vec

class Word2VecModelTrainer:
    def train_word2vec_model(self, sentences_str: str) -> str:
        try:
            sentences = [sentence.split() for sentence in sentences_str.split(";") if sentence]
            if not sentences:
                return "Training failed"

            model = Word2Vec(sentences, vector_size=50, window=3, min_count=1, workers=4)
            words = list(model.wv.index_to_key)
            return "|".join(words)
        except Exception:
            return "Training failed"
